allennlp/data/instance.py

```diff
@@ -14,16 +14,17 @@ class Instance:
     as outputs.
 
     The ``Fields`` in an ``Instance`` can start out either indexed or un-indexed.  During the data
-    processing pipeline, all fields will end up as ``IndexedFields``, and will then be converted
-    into padded arrays by a ``DataGenerator``.
+    processing pipeline, all fields will be indexed, after which multiple instances can be combined
+    into a ``Batch`` and then converted into padded arrays.
 
     Parameters
     ----------
     fields : ``Dict[str, Field]``
         The ``Field`` objects that will be used to produce data arrays for this instance.
     """
-    def __init__(self, fields: Dict[str, Field]) -> None:
+    def __init__(self, fields: Mapping[str, Field]) -> None:
```

Comment:
Why change this to `Mapping`?

Reply:
for some reason I don't understand `mypy` started complaining at me that `Dict` isn't covariant in its value type and that I should use `Mapping` instead. I don't really understand what I changed to start getting that warning, but I just did it and then mypy was happy.